### PR TITLE
Delete a full analysis with a single click

### PR DESCRIPTION
**Pull request by antgonza**
review/merge this first: ~https://github.com/biocore/qiita/pull/2402~ https://github.com/biocore/qiita/pull/2405

### CODE REVIEW

File: qiita_pet/handlers/analysis_handlers/listing_handlers.py [hunk 1]
```diff
@@ -47,34 +48,49 @@ def get(self):
...
+            job_info = r_client.get('analysis_delete_%d' % analysis_id)
+            if job_info:
+                job_info = defaultdict(lambda: '', loads(job_info))
+                job_id = job_info['job_id']
+                job = ProcessingJob(job_id)
+                job_status = job.status
+                processing = job_status not in ('success', 'error')
+                if processing:
+                    messages['info'] += (
+                        'Analysis %s in being deleted<br/>' % analysis_id)
```
Comment (josenavas) [inline qiita_pet/handlers/analysis_handlers/listing_handlers.py]: in -> is

File: qiita_ware/private_plugin.py [hunk 2]
```diff
@@ -335,6 +335,29 @@ def complete_job(job):
...
+
+        artifacts = sorted(
+            analysis.artifacts, key=lambda a: a.id, reverse=True)
+
+        for artifact in artifacts:
+            qdb.artifact.Artifact.delete(artifact.id)
+
+        qdb.analysis.Analysis.delete(analysis.id)
+
+        job._set_status('success')
```
Comment (josenavas) [inline qiita_ware/private_plugin.py]: Can you also delete the `'analysis_delete_%d'` key from redis once this job is completed? That will no longer be needed.

**Comment by antgonza** [comment]
In my original post, I forgot to add a screenshot of how the messages will appear if you are deleting multiple analyses, here we go:
screen shot 2017-11-28 at 8 24 53 am

**Comment by ElDeveloper** [comment]
Thanks @antgonza !

**Pull request merged**